File: pset1/cash.c
#include <stdio.h>
#include <cs50.h>
#include <math.h>

int main(void)
{
    // declare variables
    int change_owed;
    int coins = 0;

    // multiplying user input by 100 to get the number of cents
    // rounding the result to avoid the int being floored due to imprecision
    change_owed = round(get_float("Change owed: ") * 100);

    // while there is still change owed, subtract the largest coin possible, and increase the coins int by 1
    while (change_owed > 0)
    {
        if (change_owed >= 25)
        {
            change_owed -= 25;
            coins += 1;
        }
        else if (change_owed >= 10)
        {
            change_owed -= 10;
            coins += 1;
        }
        else if (change_owed >= 5)
        {
            change_owed -= 5;
            coins += 1;
        }
        else if (change_owed >= 1)
        {
            change_owed -= 1;
            coins += 1;
        }
    }

    printf("%i\n", coins);
}

File: pset6/cash/cash.py
from cs50 import get_float

# prompt user until non-negative number input
while(True):
    change_owed = get_float("Enter the change owed: ")
    if change_owed >= 0:
        break

# initialize variables and convert change_owed to cents
coin_list = [25, 10, 5, 1]
coins_returned = 0
change_owed *= 100

for _, coin in enumerate(coin_list):
    coins_returned += change_owed // coin
    change_owed %= coin

print(coins_returned)